src/main/NWN/API/Object/NwSound.cs

```diff
@@ -26,5 +26,11 @@ public sbyte Volume
     /// Stops this sound object from playing.
     /// </summary>
     public void Stop() => NWScript.SoundObjectStop(this);
+
+    /// <summary>
+    /// Gets the length of a wave (audio) file in seconds.
+    /// </summary>
+    public float DialogSoundLength(int strRef)
+      => NWScript.GetDialogSoundLength(strRef);
```

Comment:
This is a tricky function to categorize - I don't think this is the place for it though.

Are there any other functions that use .tlk indices?

Reply:
`don't think this is the place for it though.`
I'm not sure myself.  I made the best guess I could based on the flow chart I referenced.

`Are there any other functions that use .tlk indices?`
Again I'm not sure.  I personally never used this function in nwscript or seen it implemented.